File: DataBindingAAC/app/src/main/java/com/example/raghu/databindingaac/BindableString.java
package com.example.raghu.databindingaac;

import android.databinding.BaseObservable;

import java.util.Objects;

/**
 * Created by raghu on 3/1/18.
 */

public class BindableString extends BaseObservable {
    private String value;
    public String get() {
        return value != null ? value : "";
    }
    public void set(String value) {
        if (!Objects.equals(this.value, value)) {
            this.value = value;
            notifyChange();
        }
    }
    public boolean isEmpty() {
        return value == null || value.isEmpty();
    }
}

File: DataBindingAAC/app/src/main/java/com/example/raghu/databindingaac/BindableString.kt
package com.example.raghu.databindingaac

import androidx.databinding.BaseObservable

/**
 * Created by raghu on 3/1/18.
 */
class BindableString : BaseObservable() {
    private var value: String? = null
    fun get(): String {
        return if (value != null) value!! else ""
    }

    fun set(value: String?) {
        if (this.value != value) {
            this.value = value
            notifyChange()
        }
    }

    val isEmpty: Boolean
        get() = value == null || value!!.isEmpty()
}